### PR TITLE
build: more strict tsconfig

### PR DESCRIPTION
**Pull request by vthinkxie**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: components/progress/nz-progress-utils.ts [hunk 1]
```diff
@@ -18,13 +18,12 @@ export const sortGradient = (gradients: NzProgressGradientProgress) => {
   Object.keys(gradients).forEach(key => {
     const value = gradients[key];
     const formatKey = stripPercentToNumber(key);
-    if (isNaN(formatKey)) {
-      return {};
```
Comment (vthinkxie) [inline components/progress/nz-progress-utils.ts]: cc @wendellhu95

Comment (wendellhu95) [reply components/progress/nz-progress-utils.ts]: Reasonable.

**Comment by xiaoxiangmoe** [comment]
@vthinkxie  Why not just open 'strict: true' and close some options.

**Comment by vthinkxie** [comment]
`strict: true` contains other strict options we don't want to introduce yet.

**Comment by xiaoxiangmoe** [comment]
@vthinkxie open strict: true and close specific strict options

**Pull request merged**